build_all/base-libraries/AzureIoTHub/README.md

```diff
@@ -1,7 +1,135 @@
+
+Currently supported hardware:
+
+- ESP8266 based boards with [esp8266/arduino](https://github.com/esp8266/arduino)
+
+  - SparkFun [Thing](https://www.sparkfun.com/products/13711)
+
+  - Adafruit [Feather Huzzah](https://www.adafruit.com/products/2821)
+  
+- ESP32 based boards with [espressif/arduino-esp32](https://github.com/espressif/arduino-esp32)
+  
+  - Adafruit [HUZZAH32](https://www.adafruit.com/product/3405)
+
+## Prerequisites
+
+You should have the following ready before beginning with any board:
+
+-   [Setup your IoT hub](https://github.com/Azure/azure-iot-device-ecosystem/blob/master/setup_iothub.md)
+
+-   [Provision your device and get its credentials](https://github.com/Azure/azure-iot-device-ecosystem/blob/master/setup_iothub.md#create-new-device-in-the-iot-hub-device-identity-registry)
+
+-   [Arduino IDE](https://www.arduino.cc/en/Main/Software)
+
+-   Generate the Libraries by executing the make_sdk.cmd file with the folder to build to as arg 1, and the board you are using (esp8266, esp32) as arg 2 *OR*:
+
+	-   Install the `AzureIoTHub` library via the Arduino IDE Library Manager
+
+	-   Install the `AzureIoTUtility` library via the Arduino IDE Library Manager
+
+	-   Install the `AzureIoTProtocol_MQTT` library via the Arduino IDE Library Manager
+
+	-   Install the `AzureIoTSocket_WiFi` library via the Arduino IDE Library Manager
+
+# Simple Sample Instructions
+
+## ESP8266
+
+##### Sparkfun Thing, Adafruit Feather Huzzah, or generic ESP8266 board
+
+1. Install esp8266 board support into your Arduino IDE.
+
+    - Start Arduino and open Preferences window.
+
+    - Enter `http://arduino.esp8266.com/stable/package_esp8266com_index.json` into Additional Board Manager URLs field. You can add multiple URLs, separating them with commas.
+
+    - Open Boards Manager from Tools > Board menu and install esp8266 platform 2.5.2 or later
+
+    - Select your ESP8266 board from Tools > Board menu after installation
+
+2. Open the `iothub_II_telemetry_sample` example from the Arduino IDE File->Examples->AzureIoTHub menu.
+
+3. Update Wifi SSID/Password in `iot_configs.h`
+
+    - Ensure you are using a wifi network that does not require additional manual steps after connection, such as opening a web browser.
+
+4. Update IoT Hub Connection string in `iot_configs.h`
+
+5. Navigate to where your esp8266 board package is located, typically in `C:\Users\<your username>\AppData\Local\Arduino15\packages` on Windows and `~/.arduino15/packages/` on Linux
+	
+- Locate the board's Arduino.h `hardware/esp8266/<board package version>/cores/esp8266/` and comment out the line containing `#define round(x)`, around line 137.
+
+- Locate the board's platform.txt and add the defines -DDONT_USE_UPLOADTOBLOB -DUSE_BALTIMORE_CERT on line 73 (build.extra_flags=) 
+	
+	- Note1: Please change the CERT define to the appropriate cert define if not using the global portal.azure.com server, defines for which are laid out in `certs.c`
```

Comment:
there should probably be a link to something (probably the `CMakeLists.txt` file in the sdk) so that people know what the other options are. Something like
Here are the options [here](https://github.com/Azure/azure-iot-sdk-c/blob/master/CMakeLists.txt#L76-L78).

Reply:
Actually probably to the defines in the `certs.c` since these are compiler defines and not cmake options right?

Reply:
They are not CMAKE options, but compiler defines, yes. Also, certs.c has a descriptions on which define to use for which site. I'd consider it a pretty good place to point them to if they want to use a different cert.

Reply:
Linked to that file.